### PR TITLE
Fixed Update-user in state to use useEffect/react-hook feature/40

### LINKED ISSUE #40: Fix Update-user in state to use react hook
Within state component update-user, react hooks has to be implemented correctly.

### PR DESCRIPTION
**Pull request by ChauPham90**
It works fine when I put onSubmit and onChange function in update-user.js
When I try to split it, it sends me  **Invalid Hook Call Warning**.
Still working on splitting it. Thanks for reviewing my code.

### CODE REVIEW

File: src/state/update-user.js [hunk 1]
```diff
@@ -1,23 +1,47 @@
...
+    fetch(`https://m07.herokuapp.com/${uname}`)
+      .then(response => response.json())
+      .then(user => {
+        setUser(user)
+        setEdit(user)
+      })
+      .catch(err => console.error(err));
+  }, [uname]);
+  const onSubmit = (event) => {
+    if (event) event.preventDefault();
```
Comment (VernitaJ) [inline src/state/update-user.js]: Is if statement necessary here?

File: src/state/update-user.js [hunk 2]
```diff
@@ -1,23 +1,47 @@
...
-    },
-    mode: "cors",
-    body: JSON.stringify(user)
-  })
-    .then(response => response.json())
-    .then(() => {
-      console.log("now");
+import { useState, useEffect } from "react";
+const useUpdateUser = (initState, uname) => {
+  const [edit, setEdit] = useState(initState)
```
Comment (VernitaJ) [inline src/state/update-user.js]: Where is initState given a value?

Comment (ChauPham90) [reply src/state/update-user.js]: I passed the arguments when I called function

File: src/state/update-user.js (hunk 1 above)
Comment (ChauPham90) [reply src/state/update-user.js]: ya u r right.

**Review by NicolaWinterson — APPROVED** [review]

**Review by DeveNoor — APPROVED** [review]

**Pull request merged**